libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -2376,17 +2382,37 @@ void ScriptEngine::entityScriptContentAvailable(const EntityItemID& entityID, co
         
         // END PULL SAFEURLS FROM INTERFACE.JSON Settings
         
-        bool isInWhitelist = false;  // assume unsafe
+        // GET CURRENT DOMAIN WHITELIST BYPASS, IN CASE AN ENTIRE DOMAIN IS WHITELISTED
+        QString currentDomain = DependencyManager::get<AddressManager>()->getDomainURL().host();
+        
+        QString domainSafeIP = nodeList->getDomainHandler().getHostname();
+        QString domainSafeURL = "hifi://" + currentDomain;
         for (const auto& str : safeURLS) {
-            qCDebug(scriptengine) << whitelistPrefix << "Script URL: " << scriptOrURL << "TESTING AGAINST" << str << "RESULTS IN"
-                     << scriptOrURL.startsWith(str);
-            if (!str.isEmpty() && scriptOrURL.startsWith(str)) {
+            if(domainSafeURL.startsWith(str) || domainSafeIP.startsWith(str)) {
+                qCDebug(scriptengine) << whitelistPrefix << "Whitelist Bypassed. Current Domain Host: " 
+                    << nodeList->getDomainHandler().getHostname()
+                    << "Current Domain: " << currentDomain;
                 isInWhitelist = true;
-                qCDebug(scriptengine) << whitelistPrefix << "Script approved.";
-                break;  // bail early since we found a match
+                autoPassList = true;
+            }
+        }
+        // END CURRENT DOMAIN WHITELIST BYPASS
+
+        if (ScriptEngine::getContext() == "entity_server" || autoPassList == true) { // If running on the server or waved through, do not engage whitelist.
+            isInWhitelist = true;
+        } else {
+            for (const auto& str : safeURLS) {
+                qCDebug(scriptengine) << whitelistPrefix << "Script URL: " << scriptOrURL << "TESTING AGAINST" << str << "RESULTS IN"
+                    << scriptOrURL.startsWith(str);
+                if (!str.isEmpty() && scriptOrURL.startsWith(str)) {
+                    isInWhitelist = true;
+                    qCDebug(scriptengine) << whitelistPrefix << "Script approved.";
+                    break; // bail early since we found a match
+                }
             }
         }
-        if (!isInWhitelist) {
+
+        if (!isInWhitelist || autoPassList == false) {
```

Comment:
do you need both isInWhitelist and autoPassList?  seems like they could be combined

also, should this be !isInWhitelist && !autoPassList

Reply:
Makes sense, I've made adjustments.